### PR TITLE
Versioning updates 2 | Archive versioned objects instead of deleting

### PR DESCRIPTION
**Pull request by SmittieC**
## Description

Versioned objects should be archived upon deletion instead of being removed from the DB. Reason being that if they are removed, it will take all versions with it which mutates the experiment versions, which is incorrect.

When archiving the consent form, we replace the archived one with the default one on all experiments where this were used. In the future I think the user should choose which one to use or just manually update each experiment first. Either way, we need to tell the user which experiments will be affected.

## User Impact

Users can now safely delete objects like safety layers, consent forms, source material etc without removing it from their versioned bots.

### Demo

Peek 2024-10-15 10-50

### Docs

N/A

### CODE REVIEW

**Review by snopoke — DISMISSED** [review]

**Comment by SmittieC** [comment]
Going to merge this once the base branch is merged

**Comment by SmittieC** [comment]
I had to fix a conflict with `main`. This is ready for review

File: apps/experiments/migrations/0097_consentform_is_archived_experimentroute_is_archived_and_more.py
Comment (SmittieC) [reply apps/experiments/migrations/0097_consentform_is_archived_experimentroute_is_archived_and_more.py:38]: Please check again. Looks like the fields are on the models

File: apps/experiments/views/survey.py
Comment (SmittieC) [reply apps/experiments/views/survey.py:79]: I don't think this is necessary. From the user's perspective this should be "deleted"

File: apps/events/views.py
Comment (SmittieC) [reply apps/events/views.py:116]: Not needed, since the `VersionsObjectManagerMixin` overrides the `get_queryset` method to filter out the `is_archived=True` cases. Remember this

**Review by snopoke — APPROVED** [review]

**Pull request merged**
